### ISSUE TITLE
Handle nested properties in JSDoc @typedef

### ISSUE THREAD

**Issue by fongandrew** [Microsoft/TypeScript#20078]
**TypeScript Version:**  2.7.0-dev.20171116

**Code**

```js
/**
 * @typedef {Object} App
 * @property {String} name
 * @property {Object} icons
 * @property {String} icons.image32
 * @property {String} icons.image64
 */

/** @type {App} */
const app = {
	name: 'name',
	icons: {
		image32: 'x.png',
		image64: 'y.png' 
	}
};
```

**Expected behavior:** No error.

**Actual behavior:** Error. TypeScript treats the above typedef as defining something like this:

```ts
interface App {
  name: string;
  icons: any;
  image32: string;
  image64: string;
}
```

When what we really want is this:

```ts
interface App {
  name: string;
  icons: {
    image32: string;
    image64: string;
  }
}
```

The `@property {type} parent.nestedProp` syntax can be found here (http://usejsdoc.org/tags-property.html). TypeScript currently supports a similar syntax when used with `@param` but not with `@property`.

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/7429

**Comment by fongandrew** [comment]
@mhegazy This isn't a duplicate. #7429 concerns the use of `@param` with dotted names. This PR concerns the use of `@property`. As of 2.7.0-dev.20171116, the former works but not the latter.

**Comment by NemoStein** [comment]
Any news on this?
